### ISSUE TITLE
onUnmounted is called when there is no active component instance to be associated with.

### ISSUE THREAD

**Issue by LHofman** [prazdevs/pinia-plugin-persistedstate#101]
### Describe the bug

Version
2.1.0

Steps to reproduce

- Create a new vue project with `npm init vue@latest` (include TS and Pinia for state management)
- `cd .\vue-project\`
- `npm i`
- `npm i pinia-plugin-persistedstate`
- use the plugin with `pinia.use(piniaPluginPersistedstate);` and add `, persist: true` to the counter store
- Add this code in a non-setup component

```
import { mapWritableState } from "pinia";
import { defineComponent } from "vue";
import { useCounterStore } from "../stores/counter";

export default defineComponent({
  computed: {
    ...mapWritableState(useCounterStore, ['counter']),
    otherFn(): number {
      return this.counter;
    },
  },
});

<template>
  {{otherFn}}
</template>
```

- Run the project with `npm run dev`

What is expected?
No warnings are given in the browser

What is actually happening?
A warning is given `runtime-core.esm-bundler.js:38 [Vue warn]: onUnmounted is called when there is no active component instance to be associated with. Lifecycle injection APIs can only be used during execution of setup(). If you are using async setup(), make sure to register lifecycle hooks before the first await statement.`

### Reproduction

See above

### System Info

```Shell
System:
    OS: Windows 10 10.0.19044
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 4.14 GB / 15.85 GB
  Binaries:
    Node: 16.16.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.11 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.11.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 103.0.5060.134
    Edge: Spartan (44.19041.1266.0), Chromium (103.0.1264.77)
    Internet Explorer: 11.0.19041.1566
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/prazdevs/pinia-plugin-persistedstate/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/prazdevs/pinia-plugin-persistedstate/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by prazdevs** [comment]
Can you provide a proper repo/stackblitz for faster reproduction?

Also is this issue known for Pinia standalone, is it solely due to the plugin being added ? 
The plugin doenst use the onUnmounted itself, might be because of state subscription  (`$subscribe`)

**Comment by prazdevs** [comment]
nevermind, it's a bug. at some point i dropped the detached: true from subscription. will fix it asap

**Comment by prazdevs** [comment]
Fixed with #102  release in `2.1.1`
